### PR TITLE
[FLINK-15864][k8s] Upgrade jackson-databind dependency to 2.10.1

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

Upgrade jackson-databind dependency to 2.10.1

cc @wangyang0918 

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-kubernetes/pom.xml [hunk 1]
```diff
@@ -35,6 +35,17 @@ under the License.
 		<kubernetes.client.version>4.5.2</kubernetes.client.version>
 	</properties>
 
+	<!-- Set dependency version for transitive dependencies -->
+	<dependencyManagement>
+		<dependencies>
+			<dependency>
+				<groupId>com.squareup.okhttp3</groupId>
+				<artifactId>okhttp</artifactId>
```
Comment (zentol) [inline flink-kubernetes/pom.xml]: Seems unrelated (good idea though).

**Comment by wangyang0918** [comment]
@tillrohrmann Thanks for creating this PR.

The current version(4.5.2) of kubernetes-client has fixed problem about different version conflict of dependency `com.fasterxml.jackson`. So of cause, we could remove the `exclude` part and using transitive dependencies. 

Then if we remove the explicit dependencies and exclusions, i think we do not need to shade the `com.fasterxml.jackson` related packages anymore.

**Comment by tillrohrmann** [comment]
Good point @wangyang0918. I'll remove the relocations then.

**Comment by tillrohrmann** [comment]
Well, actually I would like to leave it like this because at the moment we don't have a unrelocated jackson dependency in `flink-dist`. I think it should be a follow-up issue to unify the differently relocated Jackson dependencies and to have a single version in Flink.

**Comment by wangyang0918** [comment]
@tillrohrmann I have checked the `flink-dist_2.11-1.11-SNAPSHOT.jar`. It seems that we have already bundle the unrelocated jackson.

```
wangyang-pc:build-target danrtsey.wy$ unzip -l lib/flink-dist_2.11-1.11-SNAPSHOT.jar | grep 'com/fasterxml/jackson' | grep -v 'shaded' | wc -l
908
```

**Comment by tillrohrmann** [comment]
I think you are mistakingly counting relocated Jackson dependencies. If you run `jar tf flink-dist_2.11-1.11-SNAPSHOT.jar| grep ^com/fasterxml/jackson | wc -l` then you will get `0`.

**Comment by tillrohrmann** [comment]
Merging this PR now.

**Comment by wangyang0918** [comment]
@tillrohrmann I think the two commands have no differences. 
I build the master branch and still could find the unrelocated `jackson` classes. Do i miss something?

```
$ jar tf lib/flink-dist_2.11-1.11-SNAPSHOT.jar | grep ^com/fasterxml/jackson | wc -l
     908
```

```
$ jar tf lib/flink-dist_2.11-1.11-SNAPSHOT.jar | grep ^com/fasterxml/jackson | head -n 10
com/fasterxml/jackson/
com/fasterxml/jackson/databind/
com/fasterxml/jackson/databind/AbstractTypeResolver.class
com/fasterxml/jackson/databind/AnnotationIntrospector$ReferenceProperty$Type.class
com/fasterxml/jackson/databind/AnnotationIntrospector$ReferenceProperty.class
com/fasterxml/jackson/databind/AnnotationIntrospector.class
com/fasterxml/jackson/databind/BeanDescription.class
com/fasterxml/jackson/databind/BeanProperty$Bogus.class
com/fasterxml/jackson/databind/BeanProperty$Std.class
com/fasterxml/jackson/databind/BeanProperty.class
```

**Comment by tillrohrmann** [comment]
Hmm this is strange. Which version of maven did you use for building Flink @wangyang0918? I think it should be `<= 3.2.5` otherwise shading might not properly work. I checked also the RC1 and I couldn't find unrelocated Jackson dependencies in it.

**Comment by wangyang0918** [comment]
@tillrohrmann You are right. The maven version `>3.2.5` could not shade the `jackson` properly. Let's unify the different relocated `jackson` dependencies in the future.
